### ISSUE TITLE
Error: connect ENOENT when start debugging extension

### ISSUE THREAD

**Issue by Foolment** [microsoft/vscode#73320]
- VSCode Version: 1.33.1
- OS Version: macOS Mojave 10.14.2

Steps to Reproduce:

1. Open Visual Studio Code with any custom extension project.
2. Press <kbd>F5</kbd> to start debugging.
3. the debugging VSCode will exit automatically after a while and left the error message:
```
Error: connect ENOENT /var/folders/dc/l3lc92l976ddd8djpjmh_ssxnkcfdj/T/vscode-ipc-e95c4abd-e140-4270-9de1-592620f3b88c.sock\\n    at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1161:14)
```
![image](https://user-images.githubusercontent.com/3321518/57205925-9ab88c80-6ff4-11e9-88d1-68c9857c8619.png)
Also the .sock file does not exist.

Does this issue occur when all extensions are disabled?: Yes

**Comment by isidorn** [comment]
Can not reproduce.
Can you reproduce with vscode insiders https://code.visualstudio.com/insiders/
If yes, can you please create a repository on github which I can clone and reproduce this?

**Comment by Foolment** [comment]
I can reproduce the problem with vscode insiders or reinstall the vscode.
This is the verbose log from `code --verbose` command. Maybe do some help.
[verbose.log](https://github.com/microsoft/vscode/files/3150867/verbose.log)

I don't know how to create a repository to let you reproduce the problem, can you please show me some steps?

**Comment by weinand** [comment]
@isidorn the error looks unrelated to debugging. Please involve someone who understand the IPC socket naming.

**Comment by isidorn** [comment]
@joaomoreno did you do the ipc socket communication?

**Comment by joaomoreno** [comment]
This is the extension host socket. cc @alexandrudima

**Comment by Foolment** [comment]
Finally i solved this problem by upgrading Node from `v11.14.0` to `v12.2.0` (don't know why), thanks all guys.
